### ISSUE TITLE
Bug: TRUSTED_PROXY_IPS validation passes but TRUSTED_PROXIES has no IP format check

### ISSUE THREAD

**Issue by 5hivam123** [Checkora/Checkora#1887]
## Description
In `core/settings.py`, `TRUSTED_PROXY_IPS` has proper CIDR/IP validation 
using `ipaddress.ip_network()` and raises `ImproperlyConfigured` for invalid 
entries. However, `TRUSTED_PROXIES` (a separate setting used for client IP 
extraction) has NO such validation — it just splits the string and accepts 
any value including invalid IPs, empty strings, or garbage values.

## Steps to Reproduce
1. In `.env`, set `TRUSTED_PROXIES=999.999.999.999,not-an-ip,`
2. Start the Django development server
3. No error is raised — invalid values are silently accepted into 
   `TRUSTED_PROXIES` list

## Expected Behavior
`TRUSTED_PROXIES` should validate each entry using `ipaddress.ip_network()` 
just like `TRUSTED_PROXY_IPS` does, and raise `ImproperlyConfigured` for 
invalid values.

## Actual Behavior
Invalid IP addresses in `TRUSTED_PROXIES` are silently accepted with no 
error, potentially causing incorrect client IP extraction and broken rate 
limiting behavior.

## Screenshots
N/A — silent configuration bug.

## Environment
- OS: Windows 11
- Browser: N/A (backend/settings issue)
- Python Version: 3.12

## Additional Context
Affected file: `core/settings.py`

Current code (no validation):
```python
TRUSTED_PROXIES = os.environ.get('TRUSTED_PROXIES', '127.0.0.1,::1').split(',')
TRUSTED_PROXIES = [ip.strip() for ip in TRUSTED_PROXIES if ip.strip()]
```

`TRUSTED_PROXY_IPS` (has validation — should be replicated):
```python
for entry in _trusted_proxy_ips_raw:
    try:
        ipaddress.ip_network(entry, strict=False)
    except ValueError:
        _invalid_trusted_proxy_ips.append(entry)
```

Labels suggested: `bug`, `good first issue`, `GSSoC`

**Comment by 5hivam123** [comment]
I'd like to work on this fix myself as part of GSSoC. Can you please assign it to me? 🙏 #GSSoC

**Comment by saurabhhhcodes** [comment]
Hi @Checkora/maintainers, I've made the fix for this issue. Could I please be assigned so I can open a PR? Thanks!

**Comment by saurabhhhcodes** [comment]
Hello! I'd like to work on this issue under GSSoC 26. Please assign it to me so I can submit a fix. Thanks!
